app/controllers/products_controller.rb

```diff
@@ -1,9 +1,12 @@
 class ProductsController < ApplicationController
+  skip_before_action :authenticate_user!, only: %i[index show]
+
   def index
     @products = Product.purchasable.order(:position).page(params[:page])
   end
 
   def show
     @product = Product.purchasable.find(params[:id])
+    @cart_item = current_cart.cart_items.find_or_initialize_by(product: @product) if user_signed_in?
```

Comment:
:+1: 

好みですが
```suggestion
    @cart_item = current_cart&.cart_items&.find_or_initialize_by(product: @product)
```
とも書けます

Reply:
この書き方のほうがいいですね！修正しました
[7edf409](https://github.com/kk0000-kk/sakura-market/pull/25/commits/7edf4094343caebb22677a5a579e9674ef3a51f4)